asapdiscovery-data/asapdiscovery/data/scripts/download_fragalysis_data.py

```diff
@@ -26,6 +26,8 @@ def main():
 
     # Copy the base call and update the base target with the cli-specified target
     api_call = copy.deepcopy(API_CALL_BASE)
+    if args.target.lower() not in {"mpro", "mac1"}:
+        raise ValueError("Target must be one of [mpro, mac1].")
```

Comment:
I don't know if it's a good idea that we limit ourselves here, is there a special reason why we want to do this? I'm just trying to avoid having to remember adding every target manually here when we get them in fragalysis.

Reply:
When the target is not found on fragalysis an `HTTPError`will be raised from https://github.com/choderalab/asapdiscovery/blob/6ed0ed76ef830ab23a26dd58a7af759a557916b2/asapdiscovery-data/asapdiscovery/data/fragalysis.py#L49 just in case that was the reason

Reply:
This isn't a blocker for me, but just wanted to mention it.

Reply:
yeah it's a fair point. my motivation is just that the `HTTPError` doesn't give a reason for the failure, just that it fails. eg:
```python
Traceback (most recent call last):
  File "/home/kaminowb/miniconda3/envs/asapdiscovery_dev/bin/download-fragalysis-data", line 8, in <module>
    sys.exit(main())
  File "/lila/data/chodera/kaminowb/moonshot_ml_dev/asapdiscovery_dev/asapdiscovery-data/asapdiscovery/data/scripts/download_fragalysis_data.py", line 33, in main
    download(args.o, api_call, args.x)
  File "/lila/data/chodera/kaminowb/moonshot_ml_dev/asapdiscovery_dev/asapdiscovery-data/asapdiscovery/data/fragalysis.py", line 49, in download
    raise requests.HTTPError(
requests.exceptions.HTTPError: Post request to https://fragalysis.diamond.ac.uk/api/download_structures/ failed with 404 error code, using the following API call {'target_name': 'Sdf', 'proteins': '', 'event_info': False, 'sigmaa_info': False, 'diff_info': False, 'trans_matrix_info': False, 'NAN': False, 'mtz_info': False, 'cif_info': False, 'NAN2': False, 'map_info': False, 'single_sdf_file': True, 'sdf_info': True, 'pdb_info': False, 'bound_info': True, 'metadata_info': True, 'smiles_info': True, 'static_link': False, 'file_url': ''}.
```
as you say though, it might not be good to limit, especially since there are other targets on Fragalysis that we're not working with

Reply:
I see, I don't have a strong opinion about this, I think we can have this limit in the meantime. Just wanted to mention that we are returning the full API call used when we get the `HTTPError` hoping that users can debug it if needed.

On the other hand, I think we can have the option/choices limit directly using `ArgParser`, which I think it's a bit more elegant. I'll be making those changes soon.

Reply:
Done in https://github.com/choderalab/asapdiscovery/pull/255/commits/732ad26b9e6f79ac8e693f8dbf9b2fd58672bacd

Reply:
oh perfect, thanks!